### PR TITLE
First run

### PR DESCRIPTION
**Pull request by swstack**
Hey guys,

I decided to check out the scitran project and started working from the developer docs in the readme.  I figured i'd drop a PR with some improvements (IMO) for controlling your development environment and getting up and running quickly.  My commits are very granular, maybe more so I would do normally, but I wanted to call out everything I was doing.

Disclaimer: other than verifying that `./scitran.py start` doesn't crash on import errors or the like, I haven't done any testing with these changes but they should "just work" assuming there is no craziness in build scripts, etc.

This PR features the following high-level changes:

* Isolate python environment using virtualenv
* Capture python dependencies in requirements.txt used by pip, eventually you should freeze dependency versions if the application is live.
*  lib/ doesn't (i don't think) need to be on the path anymore, even though i kept it in the code just in case

I hope what I've done here is helpful!  I will likely try to get an instance up and running later (I'm in class today so I did this over lunch :P)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,8 +29,7 @@ If you plan to work on SciTran, clone this repository and install our dependenci
 git clone https://github.com/scitran/scitran.git && cd scitran
 
 sudo apt-get install -y python-pip
-sudo pip install --upgrade  pip
-sudo pip install --target lib --upgrade docker-py requests sh toml
+./bootstrap.sh
```
Comment (kofalt) [inline README.md]: It looks like your added script is actually `./pybootstrap.sh`

File: README.md [hunk 2]
```diff
@@ -50,6 +49,12 @@ git clone https://github.com/scitran/testdata.git code/testdata
 
 ```
 
+One-liner:
+
+```
+git clone https://github.com/scitran/api.git code/api; git clone https://github.com/scitran/data.git code/data; git clone https://github.com/scitran/sdm.git code/sdm; git clone https://github.com/scitran/testdata.git code/testdata
+```
```
Comment (kofalt) [inline README.md]: If `pybootstrap.sh` is to be specifically a new-developer entrypoint, we may as well move these clone commands into the script.

File: pybootstrap.sh [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env sh
+
+virtualenv .env
```
Comment (kofalt) [inline pybootstrap.sh]: Is `virtualenv` included in a (modern-ish) Ubuntu machine that has only the implicit system python and `python-pip` installed from apt? If not, should we include installation instructions?

**Comment by kofalt** [comment]
Added some comments.

At a higher level, when I threw this together last week @kevlarkevin and I specifically omitted the use of a virtualenv for fear of making things too complicated, the assumption being python enthusiasts would already know how to use vEnv.

I would be very open to reconsidering that decision; I like that it's only used for the development instructions. As long as it's not too confusing to use I'm all for it.

**Comment by gsfr** [comment]
@kevlarkevin did a major overhaul of this stuff on Friday after some real-world deployment experience. He's out today, but should be back tomorrow. Not sure we'll merge this because it's overlapping and conflicting with work already done.

**Comment by swstack** [comment]
@gsfr , I totally understand if this conflicts with work already done, or if this doesn't match up with your guys' work flow.  At the very least it caused some thinking regarding the pros/cons of this approach and thinking is always a good thing!

File: README.md (hunk 2 above)
Comment (swstack) [inline README.md]: Good eye!!  If yall want to move forward with any of these changes I'll amend this commit to reflect the actual filename, thanks for reviewing!

**Comment by kofalt** [comment]
Ooh, I'm guessing that's the [new-deploy](https://github.com/scitran/scitran/commits/new_deploy) branch.

File: README.md (hunk 2 above)
Comment (kofalt) [inline README.md]: Agreed, let's pick this up tomorrow-ish after Kevin gets his branch merged, anything he didn't cover we can massage in.

**Comment by kofalt** [comment]
@kevlarkevin With #7 landed, let's either land parts of this and/or cherrypick commits for work on #12, etc. In particular `pybootstrap.sh` has some virtualenv setup that I assume can be either used directly or adapted.

**Comment by kevlarkevin** [comment]
Thanks for this.  The following ideas from this PR were added in a different PR, #15:
- use virtualenv  and add `.gitignore` rule for virtualenv
- capture dependencies using `pip freeze > requirements.txt`
- lib was removed from sys.path.

**Pull request closed without merging**